source/routing/heat_map.cpp

```diff
@@ -141,26 +141,27 @@ static std::vector<std::vector<Projection>> find_projection(BoundBox box,
     std::vector<std::vector<Projection>> dist_pixel = {step, {step, Projection()}};
     const size_t offset_lon = floor(min_dist / (width_step * N_DEG_TO_DISTANCE)) + 1;
     const size_t offset_lat = floor(min_dist / (height_step * N_DEG_TO_DISTANCE)) + 1;
-    auto begin = std::lower_bound(
-        worker.pl.items.begin(), worker.pl.items.end(), box.min.lon(),
-        [](const proximitylist::ProximityList<georef::vertex_t>::Item& i, double min) { return i.coord.lon() < min; });
 
-    if (begin == worker.pl.items.end()) {
-        return dist_pixel;
+    auto box_center = type::GeographicalCoord{(box.min.lon() + box.max.lon()) / 2, (box.min.lat() + box.max.lat()) / 2};
+
+    auto radius = box.min.distance_to(box.max);
```

Comment:
Isn't this the diameter?

Reply:
you're right, it IS a diameter. but for the NN index, it's a search radius :) 

Apparently, the radius is too small, i have to use the diameter. 
For those points that are not inside of the box, we filter them at [this line](https://github.com/CanalTP/navitia/pull/2813/files#diff-482ca655704e9c2f731b66fdf1337634L159)